utils/unittest/unittest.go

```diff
@@ -0,0 +1,96 @@
+package unittest
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/caicloud/nirvana/definition"
+	"github.com/caicloud/nirvana/service"
+)
+
+// ResponseWriter mocks http.ResponseWriter and records response for testing
+type ResponseWriter interface {
+	http.ResponseWriter
+	Code() int
+	Bytes() []byte
+}
+
+// NewResponseWriter creates a new ResponseWriter
+func NewResponseWriter() ResponseWriter {
+	return newRW()
+}
+
+type responseWriter struct {
+	code   int
+	header http.Header
+	buf    *bytes.Buffer
+}
+
+func newRW() *responseWriter {
+	return &responseWriter{0, http.Header{}, bytes.NewBuffer(nil)}
+}
+
+func (r *responseWriter) Write(d []byte) (int, error) {
+	return r.buf.Write(d)
+}
+
+func (r *responseWriter) WriteHeader(code int) {
+	r.code = code
+}
+
+func (r *responseWriter) Header() http.Header {
+	return r.header
+}
+
+func (r *responseWriter) Code() int {
+	return r.code
+}
+
+func (r *responseWriter) Bytes() []byte {
+	return r.buf.Bytes()
+}
+
+// NewTestService creates a service.Service for testing
+func NewTestService(desc ...definition.Descriptor) (service.Service, error) {
+	builder := service.NewBuilder()
+	builder.SetModifier(service.FirstContextParameter())
+	builder.AddFilter(service.RedirectTrailingSlash(), service.FillLeadingSlash(), service.ParseRequestForm())
```

Comment:
Modifiers and filters may be changed by users. So these configs needs to be exported.

Reply:
added a new function `NewTestServiceWithConfig `